main.go

```diff
@@ -0,0 +1,342 @@
+		panic(err)
+	}
+	cclient, err := client.New(*cconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a dummy profile in ETCD datastore which will be used to test calico Cache priming
+	profiles := cclient.Profiles()
+
+	_, err = profiles.Create(&api.Profile{
+		Metadata: api.ProfileMetadata{
+			Labels:       map[string]string{"tier": "uat"},
+			Name:         "newProfile",
+			Tags:	    	[]string{"vinshin"},
+		},
+		Spec: api.ProfileSpec{
+			IngressRules : []api.Rule{},
+			EgressRules  : []api.Rule{},
+		},
+	})
+
+	if err != nil {
+		glog.Error("Profile newProfile already exists on calico", err)
+	}
+
+	ccache := calicocache.NewCache(queue,cclient,"Profile")
+	
+	stop := make(chan struct{})
+	defer close(stop)
+	
+	
+	
+	
+	
+	// Bind the workqueue to a cache with the help of an informer. This way we make sure that
+	// whenever the cache is updated, the node key is added to the workqueue.
+	// Note that when we finally process the item from the workqueue, we might see a newer version
+	// of the Node than the version which was responsible for triggering the update.
+	indexer, informer := cache.NewIndexerInformer(nodeListWatcher, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			glog.Infof("Got Add for namespace %s\n", key)
+			
+			if err == nil {
+				ccache.Set(key,obj)
+			}
+		},
+		UpdateFunc: func(old interface{}, new interface{}) {
+			glog.Infof("Got update for namespace Old Ob: %#v\n", old)
+			glog.Info("\n------\n")
+			glog.Infof("New object %#v\n",new)
+			key, err := cache.MetaNamespaceKeyFunc(new)
+			if err == nil {
+				ccache.Set(key,new)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+			// key function.
+			
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			glog.Infof("Got delete for namespace %s\n", key)
+			if err == nil {
+				ccache.Delete(key)
+			}
+		},
+	}, cache.Indexers{})
+
+	ccache.SetIndexer(indexer)
+	controller := NewController(queue, ccache, cclient, indexer, informer)
+
+	// Now let's start the Kubernetes controller
+	fmt.Println("Starting controller")
+	
+	go controller.Run(5, stop)
+
+	
+	
+
+	// Wait forever.
+	select {}
+}
+
+type Controller struct {
+	indexer        cache.Indexer
+	queue          workqueue.RateLimitingInterface
+	informer       cache.Controller
+	calicoObjCache calicocache.ResourceCache
+	calicoClient   *client.Client
+}
+
+type QueueUpdate struct {
+	Key   string
+	Force bool
+}
+
+func (c *Controller) Run(threadiness int, stopCh chan struct{}) {
+	defer uruntime.HandleCrash()
+
+	// Let the workers stop when we are done
+	defer c.queue.ShutDown()
+	glog.Info("Starting Node controller")
+
+	go c.informer.Run(stopCh)
+
+	// Wait for all involved caches to be synced, before processing items from the queue is started
+	if !cache.WaitForCacheSync(stopCh, c.informer.HasSynced) {
```

Comment:
When we start Calico cache, we prime it only objects that are also present in kubernetes. 
To get this information, we will need to first start informer which will fill up kubernetes cache and then we can query on it. Otherwise we may need to connect to K8S API server for each object priming.
Let me know if there are any issues with current order of syncing.